src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -248,6 +248,34 @@ internal static class UriHelper
             return dest;
         }
 
+        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+            ref int destPosition, char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax,
+            bool isQuery)
+        {
+            fixed (char* pStr = input)
+            {
+                return UnescapeString(pStr, start, end, dest, ref destPosition, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+            }
+        }
+
+        internal static unsafe char[] UnescapeString(char* pStr, int start, int end, char[] dest, ref int destPosition,
+            char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax, bool isQuery)
+        {
+            PooledCharArray pooledArray = new PooledCharArray(dest);
```

Comment:
> new PooledCharArray(dest); [](start = 41, length = 27)

this is a new allocation. but you don't need one, you have dest to write to. are you only doing this because UnescapeString expects a pooled array? If so we should maybe think about how to avoid that.